Integrations/integration-Active_Directory_Query.yml

```diff
@@ -0,0 +1,1205 @@
+
+        args = demisto.args()
+
+        search_base = args.get('base-dn') or DEFAULT_BASE_DN
+
+        # get the dn of the member - either user or computer
+        args_err = "Pleade provide either username or computer-name"
+        member_dn = ''
+
+        if args.get('username') and args.get('computer-name'):
+            # both arguments passed
+            raise Exception(args_err)
+        if args.get('username'):
+            member_dn = user_dn(args['username'], search_base)
+        elif args.get('computer-name'):
+            member_dn = computer_dn(args['computer-name'], search_base)
+        else:
+            # none of the arguments passed
+            raise Exception(args_err)
+
+        grp_dn = group_dn(args.get('group-cn'), search_base)
+
+        success = microsoft.removeMembersFromGroups.ad_remove_members_from_groups(conn, [member_dn], [grp_dn], True)
+        if not success:
+            raise Exception("Failed to remove {member} from group {group_name}".format({
+                "member": args.get('username') or args.get('computer-name'),
+                "group_name": args.get('group_name')
+            }))
+
+        demisto_entry = {
+            'ContentsFormat': formats['text'],
+            'Type': entryTypes['note'],
+            'Contents': "Object with dn {} removed from group {}".format(member_dn, args.get('group-cn'))
+        }
+        demisto.results(demisto_entry)
+
+    def unlock_account():
+        args = demisto.args()
+
+        # get user DN
+        sam_account_name = args.get('username')
+        search_base = args.get('base-dn') or DEFAULT_BASE_DN
+        dn = user_dn(sam_account_name, search_base)
+
+        success = microsoft.unlockAccount.ad_unlock_account(conn, dn)
+        if not success:
+            raise Exception("Failed to unlock user {}".format(sam_account_name))
+
+        demisto_entry = {
+            'ContentsFormat': formats['text'],
+            'Type': entryTypes['note'],
+            'Contents': "Unlocked user {}".format(sam_account_name)
+        }
+        demisto.results(demisto_entry)
+
+    ''' DELETE OBJECT '''
+
+    def delete_user():
+        # can acually delete any object...
+        success = conn.delete(demisto.args().get('user-dn'))
+        if not success:
+            raise Exception('Failed to delete user')
+
+        demisto_entry = {
+            'ContentsFormat': formats['text'],
+            'Type': entryTypes['note'],
+            'Contents': "Deleted object with dn {}".format(demisto.args().get('user-dn'))
+        }
+        demisto.results(demisto_entry)
+
+
+    ''' EXECUTION '''
+
+    # intialize connection to LDAP server with NTLM authentication
+    server = initialize_server()
+    conn = Connection(server, user=SERVER_IP + '\\' + USERNAME, password=PASSWORD, authentication=NTLM)
+
+    try:
+        # open socket and bind to server
+        if not conn.bind():
+            demisto.results(error_entry('Failed to bind to server'))
```

Comment:
Try to get more details why the bind failed. Maybe `conn.result` can provide more insight.